app/src/main/kotlin/com/google/samples/apps/nowinandroid/NiaApplication.kt

```diff
@@ -37,10 +39,25 @@ class NiaApplication : Application(), ImageLoaderFactory {
 
     override fun onCreate() {
         super.onCreate()
+
+        // Kill NiA if there are main thread policy violations and log the offending call.
```

Comment:
The `Application.onCreate` method is a fairly sacred space and should be kept as clean as possible. Please refactor this into a separate function named `setStrictModeThreadPolicy` and add the following comment above where the policy is actually set: 

```
// Set a thread policy that detects all potential problems on the main thread, 
// such as network and disk access, and will log the offending call and kill the application.
```